code/modules/mob/living/simple_animal/parrot/_parrot.dm

```diff
@@ -0,0 +1,356 @@
+	pass_flags = PASS_FLAG_TABLE
+	mob_size = MOB_SIZE_SMALL
+
+	speak =       list("Hi","Hello!","Cracker?")
+	speak_emote = list("squawks","says","yells")
+	emote_hear =  list("squawks","bawks")
+	emote_see =   list("flutters its wings")
+
+	natural_weapon = /obj/item/natural_weapon/beak
+	speak_chance = 1//1% (1 in 100) chance every tick; So about once per 150 seconds, assuming an average tick is 1.5s
+	turns_per_move = 5
+	response_harm = "swats"
+	stop_automated_movement = 1
+	universal_speak = TRUE
+	ai = /datum/ai/parrot
+
+	meat_type = /obj/item/chems/food/meat/chicken/game
+	meat_amount = 3
+	skin_material = /decl/material/solid/skin/feathers
+
+	/// Headset for Poly to yell at engineers :)
+	var/obj/item/radio/headset/ears = null
+	/// Parrots are kleptomaniacs. This variable ... stores the item a parrot is holding.
+	var/obj/item/held_item
+	/// Maximuim w_class of a held item.
+	var/max_held_item_size = ITEM_SIZE_SMALL
+	/// Reference to our current perch.
+	var/obj/perched_on_atom
+
+/mob/living/simple_animal/parrot/Initialize()
+	// If ears is truthy, we want a headset. Either one has
+	// been supplied, or we pick one from a random list below.
+	if(ears)
+		if(!ispath(ears, /obj/item/radio/headset))
+			ears = pick(list(
+				/obj/item/radio/headset/headset_sec,
+				/obj/item/radio/headset/headset_eng,
+				/obj/item/radio/headset/headset_med,
+				/obj/item/radio/headset/headset_sci,
+				/obj/item/radio/headset/headset_cargo
+			))
+		ears = new ears(src)
+	. = ..()
+	update_icon()
+
+/mob/living/simple_animal/parrot/Destroy()
+	QDEL_NULL(ears)
+	if(held_item)
+		drop_from_inventory(held_item)
+	perch_on(null)
+	return ..()
+
+/mob/living/simple_animal/parrot/death(gibbed, deathmessage, show_dead_message)
+	if(held_item)
+		drop_from_inventory(held_item)
+	perch_on(null)
+	return ..(gibbed, deathmessage, show_dead_message)
+
+/*
+ * Perch handling.
+ */
+/mob/living/simple_animal/parrot/proc/check_perch_state()
+	if(QDELETED(perched_on_atom) || get_dist(perched_on_atom, src) > world.view)
+		perch_on(null)
+		return
+	// If it's moving and we're perched on it, stay perched if we can.
+	if(perched_on_atom && perched_on_atom.loc != get_turf(src) && Adjacent(perched_on_atom))
+		glide_size = perched_on_atom.glide_size
+		forceMove(get_turf(perched_on_atom))
+	if(istype(ai, /datum/ai/parrot))
+		var/datum/ai/parrot/parrot_ai = ai
+		if(parrot_ai.check_perch_state())
+			update_icon()
+
+// Parrots look very silly crawling around on the ground.
+/mob/living/simple_animal/parrot/DoMove(var/direction, var/mob/mover, var/is_external)
+	. = ..()
+	if(. && !is_external && resting && !incapacitated() && (!perched_on_atom || loc != get_turf(perched_on_atom)))
+		resting = FALSE
+		update_icon()
+
+/mob/living/simple_animal/parrot/proc/atom_is_perchable(var/obj/target)
+	return !perched_on_atom && istype(target) && (target.obj_flags & OBJ_FLAG_PARROT_PERCHABLE) && !(locate(/mob/living) in get_turf(target))
```

Comment:
This will stop parrots from perching on objects they're on the same tile as, which I don't think is intended.